org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/DirectoryScalaInstallation.scala

```diff
@@ -111,15 +130,33 @@ class DirectoryScalaInstallation(val directory: IPath) extends ScalaInstallation
   if (!versionCandidate.isDefined) throw new IllegalArgumentException("The Scala library jar in this directory has incorrect or missing version information, aborting.")
   // TODO : this hard-coded hook will need changing
   if (versionCandidate.isDefined && versionCandidate.get < ScalaVersion("2.10.0")) throw new IllegalArgumentException("This Scala version is too old for the presentation compiler to use. Please provide a 2.10 scala (or later).")
+  // Hydra initialization checks
+  if (isHydraInstallation) {
+    if (extraJars.filter(module => module.classJar.toFile().getName.contains(hydraBridgePrefix)).isEmpty)
+      throw new IllegalArgumentException("Can not recognize a valid Hydra Bridge jar in this directory.")
+    if (extraJars.filter(module => module.classJar.toFile().getName.contains(hydraPrefix)).isEmpty)
+      throw new IllegalArgumentException("Can not recognize a valid Hydra jar in this directory.")
+    if (extraJars.filter(module => module.classJar.toFile().getName.contains(hydraLicenseCheckingPrefix)).isEmpty)
+      throw new IllegalArgumentException("Can not recognize a valid Hydra License Checking jar in this directory.")
+    if (extraJars.filter(module => module.classJar.toFile().getName.contains(license4jPrefix)).isEmpty)
+      throw new IllegalArgumentException("Can not recognize a valid License4j jar in this directory.")
+    if (extraJars.filter(module => module.classJar.toFile().getName.contains(hydraDashboardPrefix)).isEmpty)
+      throw new IllegalArgumentException("Can not recognize a valid Hydra Dashboard jar in this directory.")
+  }
 
-  override lazy val extraJars = findScalaJars(List(scalaReflectPrefix,
-      scalaSwingPrefix), presumedLibraryVersionString).filter {
+  override lazy val extraJars = findScalaJars(List(
+    scalaReflectPrefix,
+    scalaSwingPrefix, hydraPrefix, hydraBridgePrefix, scalaLoggingPrefix,
+    logbackClassicPrefix, logbackCorePrefix, slf4jPrefix,
+    hydraReflectPrefix, scalaXmlPrefix, license4jPrefix,
+    hydraDashboardPrefix, hydraLicenseCheckingPrefix), presumedLibraryVersionString).filter {
```

Comment:
This list is something that will eventually get out of sync (one day or another we might add or remove something). Not sure if it's possible to do it some other way, but I wanted to raise the concern.